python/ml4ir/base/io/file_io.py

```diff
@@ -45,6 +45,17 @@ def make_directory(dir_path: str, clear_dir: bool = False, log=None) -> str:
     return dir_path
 
 
+def read_list(infile: str):
+    """Read a text file and convert entries into a list of strings by splitting on newline"""
+    txt = ""
+    if infile.startswith("hdfs"):
+        raise NotImplementedError
+    else:
+        txt = open(infile, "r").read()
+
+    return [token.strip() for token in txt.splitlines()]
```

Comment:
the naming assumes you are given a file with a token per line here. 
There is an alternative fancy way, for text files. You can use `pathlib` and have `infile` to be a Path. [If so](https://docs.python.org/3/library/pathlib.html#pathlib.Path.read_text): 
`infile.read_text().split("\n")` 
The advantage if that it takes care of closing the file, which is missing here.

Reply:
Did not know about this. I like it. Will do. Thanks!